### PR TITLE
Improvements to InstallationManager

### PR DESCRIPTION
**Pull request by Bond-009**
Improve public api (`IInstallationManager`)

### CODE REVIEW

File: Emby.Server.Implementations/Emby.Server.Implementations.csproj [hunk 1]
```diff
@@ -46,6 +46,11 @@
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <!-- We need C# 7.3 to compare tuples-->
+     <LangVersion>latest</LangVersion>
```
Comment (dkanada) [inline Emby.Server.Implementations/Emby.Server.Implementations.csproj]: seems to be an indentation error somewhere

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 2]
```diff
@@ -47,49 +48,15 @@ public class InstallationManager : IInstallationManager
...
         public event EventHandler<GenericEventArgs<IPlugin>> PluginUninstalled;
 
-        /// <summary>
-        /// Called when [plugin uninstalled].
-        /// </summary>
-        /// <param name="plugin">The plugin.</param>
-        private void OnPluginUninstalled(IPlugin plugin)
-        {
-            PluginUninstalled?.Invoke(this, new GenericEventArgs<IPlugin> { Argument = plugin });
-        }
```
Comment (dkanada) [inline Emby.Server.Implementations/Updates/InstallationManager.cs]: I don't see a replacement for this event anywhere?

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 3]
```diff
@@ -477,34 +427,33 @@ public async Task InstallPackage(PackageVersionInfo package, bool isPlugin, IPro
...
-
-            if (isPlugin)
-            {
-                // Set last update time if we were installed before
-                plugin = _applicationHost.Plugins.FirstOrDefault(p => string.Equals(p.Id.ToString(), package.guid, StringComparison.OrdinalIgnoreCase))
+            // Set last update time if we were installed before
+            IPlugin plugin = _applicationHost.Plugins.FirstOrDefault(p => string.Equals(p.Id.ToString(), package.guid, StringComparison.OrdinalIgnoreCase))
                            ?? _applicationHost.Plugins.FirstOrDefault(p => p.Name.Equals(package.name, StringComparison.OrdinalIgnoreCase));
-            }
+
```
Comment (dkanada) [inline Emby.Server.Implementations/Updates/InstallationManager.cs]: double newline

File: Emby.Server.Implementations/Updates/InstallationManager.cs (hunk 2 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Updates/InstallationManager.cs]: https://github.com/jellyfin/jellyfin/pull/1462/files/65a0ca2f32e2eba640fbfead46ffb5bfd02e2c88#diff-b38c8fe5bc172bada6c68b600f7cd290R574
It got inlined

**Pull request merged**